### PR TITLE
Adding a basic ip_allow test.

### PR DESCRIPTION
**Pull request by bneradt**
This adds a test to verify that ip_allow will block a method if it is configured to do so (CONNECT in this case).

### CODE REVIEW

**Comment by randall** [comment]
[approve ci autest]

File: proxy/ProxyTransaction.h [hunk 1]
```diff
@@ -83,7 +83,7 @@ class ProxyTransaction : public VConnection
 
   /// Non-Virtual Accessors
   //
-  bool is_transparent_passthrough_allowed();
+  bool is_transparent_passthrough_allowed() const;
```
Comment (randall) [inline proxy/ProxyTransaction.h:86]: seems like this shouldn't be part of this PR?

Comment (bneradt) [reply proxy/ProxyTransaction.h:86]: I fixed some const correctness issues for functions I noticed while writing the test. My intention was to be custodial, but I can remove it from here if you prefer. Those functions are exercised by the test.

**Comment by bneradt** [comment]
[approve ci autest]

**Comment by bneradt** [comment]
> Can you add tests for H2 as well?
> 
> Not only this one but I've been worried about tests that only check behavior on H1. Since this test is done with simple curl commands, it should not be difficult to test H2 as well.

HTTP/2 test added.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.
